File: web/src/lib/api/broadcast.ts
import { apiFetch } from "./client";

export type BroadcastTarget = "all_players" | "all_admins" | "sport" | "court" | "team";

export interface BroadcastPayload {
  target: BroadcastTarget;
  title: string;
  body: string;
  sport?: string;
  court_id?: string;
  team_id?: string;
  tenant_id?: string;
  city?: string;
}

export function sendBroadcast(payload: BroadcastPayload) {
  return apiFetch<{ recipient_count: number }>(`/admin/broadcast`, {
    method: "POST",
    body: JSON.stringify(payload),
  });
}

File: app/models/broadcast.py
from typing import Literal

from pydantic import BaseModel

BroadcastTarget = Literal["all_players", "all_admins", "sport", "court", "team"]


class BroadcastRequest(BaseModel):
    target: BroadcastTarget
    title: str
    body: str
    # Required depending on target (validated in broadcast_service):
    # sport -> sport, court -> court_id, team -> team_id.
    sport: str | None = None
    court_id: str | None = None
    team_id: str | None = None
    # Optional narrowing filters, usable with all_players / all_admins / sport.
    tenant_id: str | None = None
    city: str | None = None


class BroadcastResponse(BaseModel):
    recipient_count: int
